modules/utxo-lib/src/bitgo/psbt/fromHalfSigned.ts

```diff
@@ -1,145 +1,86 @@
-  if (!publicKeys || publicKeys.length !== 3) {
-    throw new Error('Invalid multisig script');
-  }
-  for (const sig of stack.slice(1)) {
-    // Ignore extra empty element re MULTISIG bug
-    if (!Buffer.isBuffer(sig) || sig.length === 0) continue;
-    const { signature, hashType } = ScriptSignature.decode(sig);
-    const hash = hashFn(hashType);
-    let signatureMatched = false;
-    for (const pubkey of publicKeys) {
-      if (eccLib.verify(hash, pubkey, signature)) {
-        signatureMatched = true;
-        partialSig.push({ pubkey, signature: sig });
-        break;
-      }
-    }
-    if (!signatureMatched) {
-      throw new Error('Invalid signature in partially signed transaction');
-    }
-  }
-  return partialSig;
+function omitUndefined<T extends Record<string, unknown>>(v: Record<string, unknown>): T {
+  return Object.fromEntries(Object.entries(v).filter(([k, v]) => v !== undefined)) as T;
 }
 
 export function getInputUpdate(
   tx: UtxoTransaction<bigint>,
   vin: number,
-  prevOut: TxOutput<bigint> & { prevTx?: Buffer }
+  prevOuts: (TxOutput<bigint> & { prevTx?: Buffer })[]
 ): PsbtInputUpdate {
-  const input = tx.ins[vin];
-  const update: PsbtInputUpdate = {};
-  if (prevOut.prevTx) {
-    update.nonWitnessUtxo = prevOut.prevTx;
+  const nonWitnessUtxo = prevOuts[vin].prevTx;
+  const { script, witness } = tx.ins[vin];
+  if (script.length === 0 && witness.length === 0) {
+    return nonWitnessUtxo ? { nonWitnessUtxo } : {};
   }
-  let redeemScript;
-  if (input.script && input.script.length) {
-    const decompiledScriptSig = bscript.decompile(input.script);
-    if (!decompiledScriptSig) {
-      throw new Error('Invalid scriptSig, failed to decompile');
-    }
-    if ((!input.witness || !input.witness.length) && decompiledScriptSig.length === 2) {
-      throw new Error();
-    }
-    redeemScript = decompiledScriptSig.pop();
-    if (!Buffer.isBuffer(redeemScript)) {
-      throw new Error('Invalid redeem script');
-    }
-    update.redeemScript = redeemScript;
-    if (opcodes.OP_0 !== redeemScript[0] || ![22, 34].includes(redeemScript.length)) {
-      // P2SH -> 2-of-3
-      const hashFn = (hashType) => tx.hashForSignature(vin, redeemScript, hashType, prevOut.value);
-      update.partialSig = matchSignatures(hashFn, redeemScript, decompiledScriptSig);
-      return update;
-    }
+
+  const parsed = parseSignatureScript(tx.ins[vin]);
+
+  function getPartialSigs(): PartialSig[] {
+    return getSignaturesWithPublicKeys(tx, vin, prevOuts, parsed.publicKeys).flatMap((signature, i) =>
+      signature
+        ? [
+            {
+              pubkey: parsed.publicKeys[i],
+              signature,
+            },
+          ]
+        : []
+    );
   }
-  if (input.witness && input.witness.length) {
-    const witness = input.witness;
-    if (witness.length > 2 && witness[witness.length - 1][0] === 0x50) {
-      throw new Error('Annex not supported');
```

Comment:
Do we want to keep this behavior of throwing if an annex is present?

Reply:
`parseSignatureScript()` is pretty strict will throw for unrecognized taproot date